### PR TITLE
mysql: Add script processor to fix grok behaviour of ES 8.7.x+

### LINKED ISSUE #6016: MYSQL/IIS log file issue in Integrations
There have been some recent failures with IIS/MYSQL have gone unnoticed after 8.7.0. (Since CI/CD is broken). 
These issues came up in beats and same can be observed in Integrations as well. 

The issue is seen since ES has updated the grok processing(8.7.0 onwards) and it is generating different outputs (Expected json files). 

`Files now report duplicate values (list) in contrast to a string value before.
`

### For MYSQL:
 [Maria DB Log](https://github.com/elastic/beats/blob/ca13ce86cf08d3aae43789650819cac33632cf6e/filebeat/module/mysql/slowlog/test/mariadb-10.3.13.log)

**Issue**: {"root['mysql.slowlog.schema']": {'`old_type': <class 'str'>, 'new_type': <class 'list'>, 'old_value': 'employees-test', 'new_value': ['employees-test', 'employees-test']}}}`

[MYSQL Ubuntu Logs
](https://github.com/elastic/beats/blob/ca13ce86cf08d3aae43789650819cac33632cf6e/filebeat/module/mysql/slowlog/test/mysql-ubuntu-8.0.15.log)

**Issue**: {"root['mysql.thread_id']": {'`old_type': <class 'str'>, 'new_type': <class 'list'>, 'old_value': '16', 'new_value': [16, '16']`}}}

[Percona Ubuntu Logs
](https://github.com/elastic/beats/blob/ca13ce86cf08d3aae43789650819cac33632cf6e/filebeat/module/mysql/slowlog/test/percona-ubuntu-8.0.15.log)

**Issue**: {"root['mysql.slowlog.schema']": {`'old_type': <class 'str'>, 'new_type': <class 'list'>, 'old_value': 'employees', 'new_value': ['employees', 'employees'`]}}}

### For IIS:

[IIS Logs 7.5
](https://github.com/elastic/beats/blob/main/filebeat/module/iis/access/test/test-iis-7.5.log)

Issue:  {"root['destination.address']": `{'old_type': <class 'str'>, 'new_type': <class 'list'>, 'old_value': '10.100.220.70', 'new_value': ['10.100.220.70', '10.100.220.70']}
`

The issue has been fixed in beats. 
But since integrations don't follow a release cycle, we need to figure out a way to fix this in integrations. 

1. Are changes required in the pipeline?
2. Does ES need to backport the change?
3. Do we need to update the Kibana version of the integrations?

The issue raised in beats: https://github.com/elastic/beats/issues/35133

### PR DESCRIPTION
**Pull request by shmsr**
- Bug

## What does this PR do?

Changes deduplicate the list and only keep the unique elements. After this, if only 1 element remains, then it's added to the result as a single element (a string) and if not, the list is kept in the result with unique elements. This handling is done to manage the new behaviour of the grok processor in 8.7.x+. Read the related issue for more details.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [ ] I have added an entry to my package's `changelog.yml` file.
- [x] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## Related issues

- Closes #6016

### CODE REVIEW

**Comment by shmsr** [comment]
cc: @HiDAl I've seen you work on similar issues in the past. Could you please also take a look?

**Comment by HiDAl** [comment]
Hey @shmsr, just to be clear: I haven't worked previously on fixing this kind of issue, I just changed the behavior of the Grok processor elastic/elasticsearch#92586.

Now, I don't know the whole business logic (Beats side, since I'm not part of the team), but looking at the code your code does not have a consistent output. In case `thread_id` is different at any point in the log lines, the processor will return a list of different `thread_id`s, which is not tested (I don't know whether is possible to have distinct thread_id per log block, though). With your code change, you've just made the current test pass. In case the `thread_id` might be different within a log block, I think this change is better: https://github.com/elastic/beats/pull/35276/files

File: packages/mysql/data_stream/slowlog/elasticsearch/ingest_pipeline/default.json [hunk 1]
```diff
@@ -32,6 +32,30 @@
         "ignore_missing": true
       }
     },
+    {
+      "script": {
```
Comment (andrewkroh) [inline packages/mysql/data_stream/slowlog/elasticsearch/ingest_pipeline/default.json:36]: Please add a `description` and a `tag` to the script processors. This helps in debugging.

Comment (shmsr) [reply packages/mysql/data_stream/slowlog/elasticsearch/ingest_pipeline/default.json:36]: Sure @andrewkroh!

**Comment by ishleenk17** [comment]
@shmsr : You can add the tests for this change and update the expected json file.

**Comment by ishleenk17** [comment]
> Hey @shmsr, just to be clear: I haven't worked previously on fixing this kind of issue, I just changed the behavior of the Grok processor [elastic/elasticsearch#92586](https://github.com/elastic/elasticsearch/pull/92586).
> 
> Now, I don't know the whole business logic (Beats side, since I'm not part of the team), but looking at the code your code does not have a consistent output. In case `thread_id` is different at any point in the log lines, the processor will return a list of different `thread_id`s, which is not tested (I don't know whether is possible to have distinct thread_id per log block, though). With your code change, you've just made the current test pass. In case the `thread_id` might be different within a log block, I think this change is better: https://github.com/elastic/beats/pull/35276/files

In the PR shared in this comment, for any new entry of thread_id in the grok, a change is needed in the painless script. 
I suppose the change made here should be fine. It returns list of distinct values when thread ids are different and a single value if thread id is same.

**Comment by shmsr** [comment]
> @shmsr : You can add the tests for this change and update the expected json file.

@ishleenk17 I did add them but had to remove them because of unavailability of setting kibana version for specific tests. Here: https://github.com/elastic/integrations/pull/6531/commits/3a656c9faeae46ac54d722d1409803b9542431fe

**Review by ishleenk17 — APPROVED** [review]
Looks good!

**Pull request merged**